### PR TITLE
Add Due Date Test

### PR DESCRIPTION
**Pull request by weichang18**
Added due date test class
Updated the validation regex for date and time
Amend comments for ModuleTest.java for yi heng

### CODE REVIEW

File: src/main/java/seedu/address/model/assignment/DueDate.java [hunk 1]
```diff
@@ -48,23 +50,27 @@ public DueDate(String date, String time) {
      */
     public DueDate(String date) {
         requireNonNull(date);
-        // TODO: use Validation regex to verify dateTime format
-        //checkArgument(isValidPhone(phone), MESSAGE_CONSTRAINTS);
+        checkArgument(isValidDate(date), MESSAGE_CONSTRAINTS);
         this.date = LocalDate.parse(date, PARSE_DATE_FORMAT);
         this.time = LocalTime.parse("2359", PARSE_TIME_FORMAT);
```
Comment (Droffilc13) [inline src/main/java/seedu/address/model/assignment/DueDate.java:55]: Perhaps you might consider creating a constant for ```"2359"```? For example, ```LATEST_TIME_IN_DAY```.

File: src/main/java/seedu/address/model/assignment/DueDate.java [hunk 2]
```diff
@@ -9,13 +10,14 @@
 
 public class DueDate {
     public static final String MESSAGE_CONSTRAINTS = "Due dates should be in a format dd/MM/yyyy HHmm";
+    public static final String DATE_VALIDATION_REGEX =
+            "^([1-9]|[0-2][0-9]|(3)[0-1])(/)([1-9]|((0)[0-9])|((1)[0-2]))(/)\\d{4}$";
```
Comment (Droffilc13) [inline src/main/java/seedu/address/model/assignment/DueDate.java:14]: Perhaps you can consider validation of dates such as ```30/02/2019```?

File: src/main/java/seedu/address/model/assignment/DueDate.java [hunk 3]
```diff
@@ -9,13 +10,14 @@
...
+    public static final String TIME_VALIDATION_REGEX = "^(00|[0,1][0-9]||2[0-3])([0-5][0-9])$";
```
Comment (Droffilc13) [inline src/main/java/seedu/address/model/assignment/DueDate.java:15]: Is there an extra ```|``` in this string at ```...[0-9]||2[0-3]...```?

**Review by Droffilc13 — CHANGES_REQUESTED** [review]
Overall looks good! Just a few suggestions for you to consider.

File: src/main/java/seedu/address/model/assignment/DueDate.java (hunk 1 above)
Comment (weichang18) [reply src/main/java/seedu/address/model/assignment/DueDate.java:55]: Good point!

File: src/main/java/seedu/address/model/assignment/DueDate.java (hunk 2 above)
Comment (weichang18) [reply src/main/java/seedu/address/model/assignment/DueDate.java:14]: The validation of dates already take into account of that test case, I will update the test file to include that

File: src/main/java/seedu/address/model/assignment/DueDate.java (hunk 3 above)
Comment (weichang18) [reply src/main/java/seedu/address/model/assignment/DueDate.java:15]: Good catch!

**Review by Droffilc13 — APPROVED** [review]
LGTM

**Pull request merged**